use/wordcount-example.md

```diff
@@ -0,0 +1,413 @@
+```
+
+Below is the code for `AddTimestampFn`, a `DoFn` invoked by `ParDo`, that sets the data element of the timestamp given the element itself. For example, if the elements were log lines, this `ParDo` could parse the time out of the log string and set it as the element's timestamp. There are no timestamps inherent in the works of Shakespeare, so in this case we've made up random timestamps just to illustrate the concept. Each line of the input text will get a random associated timestamp sometime in a 2-hour period.
+
+```
+/**
+   * Concept #2: A DoFn that sets the data element timestamp. This is a silly method, just for
+   * this example, for the bounded data case. Imagine that many ghosts of Shakespeare are all 
+   * typing madly at the same time to recreate his masterworks. Each line of the corpus will 
+   * get a random associated timestamp somewhere in a 2-hour period.
+   */
+  static class AddTimestampFn extends DoFn<String, String> {
+    private static final Duration RAND_RANGE = Duration.standardHours(2);
+    private final Instant minTimestamp;
+
+    AddTimestampFn() {
+      this.minTimestamp = new Instant(System.currentTimeMillis());
+    }
+
+    @ProcessElement
+    public void processElement(ProcessContext c) {
+      // Generate a timestamp that falls somewhere in the past two hours.
+      long randMillis = (long) (Math.random() * RAND_RANGE.getMillis());
+      Instant randomTimestamp = minTimestamp.plus(randMillis);
+      /**
+       * Set the data element with that timestamp.
+       */
+      c.outputWithTimestamp(c.element(), new Instant(randomTimestamp));
+    }
+  }
+```
+
+### Windowing
+
+Beam uses a concept called **Windowing** to subdivide a `PCollection` according to the timestamps of its individual elements. `PTransforms` that aggregate multiple elements, process each `PCollection` as a succession of multiple, finite windows, even though the entire collection itself may be of infinite size (unbounded).
+
+The `WindowingWordCount` example applies fixed-time windowing, wherein each window represents a fixed time interval. The fixed window size for this example defaults to 1 minute (you can change this with a command-line option). 
+
+```
+/**
+ * Concept #3: Window into fixed windows. The fixed window size for this example defaults to 1
+ * minute (you can change this with a command-line option).
+ */
+PCollection<String> windowedWords = input
+  .apply(Window.<String>into(
+    FixedWindows.of(Duration.standardMinutes(options.getWindowSize()))));
+```
+
+### Reusing PTransforms over windowed PCollections
+
+TODO - explain
+
+```
+/**
+ * Concept #4: Re-use our existing CountWords transform that does not have knowledge of
+ * windows over a PCollection containing windowed values.
+ */
+PCollection<KV<String, Long>> wordCounts = windowedWords.apply(new WordCount.CountWords());
+```
+
+## Write Results to BigQuery - do we want such a section since this is specific only to Dataflow?
```

Comment:
We need a section on writing to an unbounded sink in order for the example to work. Just use BQ as an example unbounded sink for now.